WW-4974 monitor not-monitored loaded files on demand

yasserzamani opened the pull request.

lukaszlenart commented on core/src/main/java/com/opensymphony/xwork2/util/fs/DefaultFileManager.java:
```diff
@@ -40,14 +40,30 @@
+            synchronized (files) {
+                for (String fileName :
+                        files.keySet()) {
+                    if (null == files.get(fileName)) {
```
Hm... right now when I saw this I think it's really doesn't make sense to put `null`s in the `files` - you will get exactly the same behaviour when the list will be empty. So this basically means (in the `monitorFile` below)

```java
        if (!reloadingConfigs) {
            return;
        }
```

instead of

```java
        if (!reloadingConfigs) {
            files.put(fileName, null);
            return;
        }
```

yasserzamani commented:
Thanks a lot! I will fix these but fortunately I remembered fileUrl sometimes is hardly coupled to underlying container e.g. IBM app servers. So 'new URL(fileName)' I guess throw exception for such container. In this PR we loose container provided url forever. This doesn't work for such situations. I should think about an another solution or fix this :)

yasserzamani commented:
Sorry didn't get the idea. Why URI while monitorFile and even other methods input are URL all. I think no need to convert to URI and increasing risk. I just need to cache input as is for possible future monitoring demand.

lukaszlenart commented on core/src/main/java/com/opensymphony/xwork2/util/fs/DefaultFileManager.java:
```diff
@@ -40,14 +40,25 @@
 
-    protected static Map<String, Revision> files = Collections.synchronizedMap(new HashMap<String, Revision>());
+    protected static final Map<String, Revision> files = Collections.synchronizedMap(new HashMap<String, Revision>());
+    private static final List<URL> lazyMonitoredFilesCache = Collections.synchronizedList(new ArrayList<URL>());
```
`equals` implementation of `URL` will resolved hosts to compare if they have the same IP, when comparing two URLs two requests are made to the external world. Maybe this is not the case here (we use files) but I'm trying to avoid using `URL`s directly.

yasserzamani commented:
Good catch but I didn't replaced files Map. I added a List and used it's add method to save urls for future possible monitoring demand. I think add method adds as an object without calling equal method. Maybe you thought I have replaced files map of strings with map of urls? No, I have kept it as is correct and added lazy monitoring privately.

lukaszlenart commented:
Ok you are right, LGTM 👍
